Cilsil/Cil/Parsers/CallParser.cs

```diff
@@ -31,26 +31,48 @@ internal class CallParser : InstructionParser
                     return false;
             }
 
+            var instrs = new List<SilInstruction>();
+
             var calledMethod = instruction.Operand as MethodReference;
-            CreateMethodCall(state,
-                             isVirtual,
-                             calledMethod,
-                             out var retTypeRef,
-                             out var retId,
-                             out var callArgs,
-                             out var callInstr);
+            if (calledMethod.GetCompatibleFullName()
+                            .Contains("System.Void System.Threading.Monitor::Enter"))
+            {
+                state.ProcDesc.PdAttributes.IsCSharpSynchronizedMethod = true;
+                instrs.Add(CreateLockedAttributeCall(true, calledMethod.Parameters.Count, state));
+            }
+            else if (calledMethod.GetCompatibleFullName()
+                                 .Contains("System.Void System.Threading.Monitor::Exit"))
+            {
+                instrs.Add(CreateLockedAttributeCall(false, calledMethod.Parameters.Count, state));
+            }
+            else
+            {
+                CreateMethodCall(state,
+                                isVirtual,
+                                calledMethod,
+                                out var retTypeRef,
+                                out var retId,
+                                out var callArgs,
+                                out var callInstr);
 
-            var instrs = new List<SilInstruction> { callInstr };
+                instrs.Add(callInstr);
 
-            // Deref on the object calling in the case of instance method, for null
-            // validation on it by Infer. Object "this" is the first argument in the argument list
-            // for instance methods.
-            if (calledMethod.HasThis && calledMethod.Name != Identifier.ConstructorIdentifier)
-            {
-                var thisArg = callArgs.First();
-                if (thisArg.Expression is VarExpression varExpression)
+                // Deref on the object calling in the case of instance method, for null
+                // validation on it by Infer. Object "this" is the first argument in the argument list
+                // for instance methods.
+                if (calledMethod.HasThis && calledMethod.Name != Identifier.ConstructorIdentifier)
                 {
-                    instrs.Insert(0, CreateDereference(varExpression, thisArg.Type, state));
+                    var thisArg = callArgs.First();
+                    if (thisArg.Expression is VarExpression varExpression)
+                    {
+                        instrs.Insert(0, CreateDereference(varExpression, thisArg.Type, state));
+                    }
+                }
+
+                var returnType = Typ.FromTypeReference(retTypeRef);
+                if (!(returnType is Tvoid))
```

Comment:
nit: use _is not_ if supported

Reply:
I think _is not_ only works for null? https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/is